libc/src/__support/FPUtil/BasicOperations.h

```diff
@@ -58,6 +58,102 @@ LIBC_INLINE T fmax(T x, T y) {
   }
 }
 
+template <typename T, cpp::enable_if_t<cpp::is_floating_point_v<T>, int> = 0>
+LIBC_INLINE T fmaximum(T x, T y) {
+  FPBits<T> bitx(x), bity(y);
+
+  if (bitx.is_nan())
+    return x;
+  if (bity.is_nan())
+    return y;
+  if (bitx.sign() != bity.sign())
+    return (bitx.is_neg() ? y : x);
+  return x > y ? x : y;
+}
+
+template <typename T, cpp::enable_if_t<cpp::is_floating_point_v<T>, int> = 0>
+LIBC_INLINE T fminimum(T x, T y) {
+  const FPBits<T> bitx(x), bity(y);
+
+  if (bitx.is_nan())
+    return x;
+  if (bity.is_nan())
+    return y;
+  if (bitx.sign() != bity.sign())
+    return (bitx.is_neg()) ? x : y;
+  return x < y ? x : y;
+}
+
+template <typename T, cpp::enable_if_t<cpp::is_floating_point_v<T>, int> = 0>
+LIBC_INLINE T fmaximum_num(T x, T y) {
+  FPBits<T> bitx(x), bity(y);
+
+  if (bitx.is_nan())
```

Comment:
I'm sorry for a late suggestion, but I just noticed that in section F.10.9.5 in C23 standard about exceptional handling of `f*_num` functions is different from their non-num versions:
- if both are `NaN`, then a quiet NaN is returned.
- if at least one of them is signaling NaN, then `FE_INVALID` is raised.

Do you mind updating `fmaximum_num` and `fminimum_num` implementations here to take care of those cases, and updating `f*_num_test` (4 of them in total) accordingly.

Thanks,

Reply:
Yea I’ll get it done tomorrow. 
are the other functions correct? I tried to follow the doc you mentioned on the issue and also looked at some gcc docs.  I’ll double check tomorrow.

Reply:
The others look correct to me.

Reply:
Here’s the definition I wa
S basing it on 
```
The fmaximum_num function returns the greater of the two values x and y. If one argument is a number and the other is a NaN, even a signaling NaN, the number is returned. Positive zero is treated as greater than negative zero.
```
I took this from here https://www.gnu.org/software/libc/manual/html_node/Misc-FP-Arithmetic.html

Reply:
@lntue hello, do changed the code. 

can you please give an idea of how to test the exception? I’m the codebase I have found examples of `EXPECT_FP_EQ_WITH_EXCEPTION`
But not sure how to use it

Reply:
Something like `EXPECT_FP_EQ_WITH_EXCEPTION(aNaN, func(aNaN, sNaN), FE_INVALID);` should work.  And you'll neet to test with all combinations of NaNs.

Reply:
Is my handling of the case associated with raising the exception correct? Because I used you example and I’m getting `which is 0 to be greater or equal to 4`. I also tried to return a quiet nan but same thing happened

Reply:
I figured it out. i found the bug